images/metering-ansible-operator/roles/meteringconfig/tasks/configure_presto_tls.yml

```diff
@@ -0,0 +1,146 @@
+#
+- name: Setup the root certificate authority
+  block:
+  - name: Generate a RSA private key for the CA
+    openssl_privatekey:
+      size: 2048
+      type: RSA
+      path: "{{ certificates_dir.path }}/ca.key"
+
+  - name: Generate a CSR for the CA
+    openssl_csr:
+      path: "{{ certificates_dir.path }}/ca.csr"
+      privatekey_path: "{{ certificates_dir.path }}/ca.key"
+      common_name: Presto Root CA
+      basicConstraints:
+      - 'CA:TRUE'
+      basicConstraints_critical: true
+
+  - name: Generate a certificate for the CA
+    openssl_certificate:
+      path: "{{ certificates_dir.path }}/ca.crt"
+      privatekey_path: "{{ certificates_dir.path }}/ca.key"
+      csr_path: "{{ certificates_dir.path }}/ca.csr"
+      provider: selfsigned
+      selfsigned_digest: sha256
+  when: meteringconfig_spec.presto.spec.presto.config.tls.enabled or meteringconfig_spec.presto.spec.presto.config.auth.enabled
+
+#
+# Generate a Server cert/key
+#
+- name: Setup the server certificate/key
+  block:
+  - name: Generate an RSA private key for Presto
+    openssl_privatekey:
+      size: 2048
+      type: RSA
+      path: "{{ certificates_dir.path }}/server.key"
+    when: meteringconfig_spec.presto.spec.presto.config.tls.key == ""
+
+  - name: Generate the Presto server CSR
+    openssl_csr:
+      path: "{{ certificates_dir.path }}/server.csr"
+      privatekey_path: "{{ certificates_dir.path }}/server.key"
+      common_name: "*.presto-nodes"
+      subject_alt_name: "DNS:*.presto-nodes.{{ meta.namespace }}.svc.cluster.local,DNS:presto,DNS:*.presto-nodes"
+      basicConstraints:
+      - 'CA:FALSE'
+      basicConstraints_critical: false
+      key_usage:
+      - digitalSignature
+      - keyEncipherment
+      extended_key_usage:
+      - serverAuth
+
+  - name: Generate the Presto server certificate
+    openssl_certificate:
+      path: "{{ certificates_dir.path }}/server.crt"
+      privatekey_path: "{{ certificates_dir.path }}/server.key"
+      csr_path: "{{ certificates_dir.path }}/server.csr"
+      provider: ownca
+      ownca_path: "{{ certificates_dir.path }}/ca.crt"
+      ownca_privatekey_path: "{{ certificates_dir.path }}/ca.key"
+      selfsigned_digest: sha256
+    when: meteringconfig_spec.presto.spec.presto.config.tls.certificate == ""
+  when: meteringconfig_ansible_generate_presto_tls_secret
+
+#
+# Generate the Client cert/key to enable client-side authentication
+#
+- name: Setup the client certificate/key
+  block:
+  - name: Generate an RSA private key for the Presto client
+    openssl_privatekey:
+      size: 2048
+      type: RSA
+      path: "{{ certificates_dir.path }}/client.key"
+    when: meteringconfig_spec.presto.spec.presto.config.auth.key == ""
+
+  - name: Generate the Presto client CSR
+    openssl_csr:
+      path: "{{ certificates_dir.path }}/client.csr"
+      privatekey_path: "{{ certificates_dir.path }}/client.key"
+      common_name: "coordinator.presto-nodes"
```

Comment:
Technically this is currently being used by all clients connecting to presto. Let's call this "presto-clients", until we move to using a client cert for each component talking to Presto.